### PR TITLE
Add enable/disable functionality to Interaction

### LINKED ISSUE #696: Interactions should have disable() and enable() calls
There should be some mechanism by which Interactions can be temporarily disabled.
We might also consider having some mechanism to disconnect Interactions.

### PR DESCRIPTION
**Pull request by jtlan**
For temporarily disabling an `Interaction` without permanently disconnecting it from the `Component` it was attached to.

API:
```Typescript
/**
 * Gets whether this Interaction is enabled.
 */
public enabled(): boolean;
/**
 * Enables or disables this Interaction.
 */
public enabled(enabled: boolean): Interaction;
```

Example fiddle: http://jsfiddle.net/j0fnbmvr/

Open questions: Should interactive `Component`s (like `DragBoxLayer`) implement this functionality as well? Or is giving access to the internal `Interaction` sufficient?

Close #696.

### CODE REVIEW

File: src/interactions/interaction.ts [hunk 1]
```diff
@@ -42,12 +48,42 @@ export class Interaction {
...
+      this._componentAttachedTo.offAnchor(this._anchorCallback);
+    }
+  }
 
+  /**
+   * Gets whether this Interaction is enabled.
+   */
+  public enabled(): boolean;
+  /**
+   * Enables or disables this Interaction.
```
Comment (bluong) [inline src/interactions/interaction.ts]: Probably document the fact that this method can chain

File: src/interactions/interaction.ts [hunk 2]
```diff
@@ -24,17 +25,22 @@ export class Interaction {
...
+    this._detach();
     this._componentAttachedTo = component;
-    component.onAnchor(this._anchorCallback);
-
+    this._attach();
     return this;
   }
 
   /**
+   * Attaches to existing _componentAttachedTo if enabled.
```
Comment (bluong) [inline src/interactions/interaction.ts]: The docs point to a private instance variable which sounds.. a bit incorrect.  Javascript users now might wonder what this `_componentAttachedTo` is and then run into trouble interacting with it.

Let's just use english here.  This also helps us if we decide to change the name of the variable later down the road

**Comment by bluong** [comment]
Documentation issues.

File: src/interactions/interaction.ts (hunk 2 above)
Comment (jtlan) [inline src/interactions/interaction.ts]: The variable is `protected` so we can't change the name without causing a break.

Technically it's attached to the Component, just not listening...

Comment (bluong) [inline src/interactions/interaction.ts]: Oh I just realized that this is documentation on a private function...  Never mind, I think? this is fine.  Not sure, but I think this is ok.

**Comment by jtlan** [comment]
Probably `DragBoxLayer` should have its own implementation -- right now if you set `resizable(true)` and `enabled(false)`, you still get the "`<-->`" cursor.

File: src/interactions/interaction.ts (hunk 1 above)
Comment (jtlan) [inline src/interactions/interaction.ts]: done.

File: src/interactions/interaction.ts [hunk 3]
```diff
@@ -24,16 +25,18 @@ export class Interaction {
...
-
+    this._disconnect();
     this._componentAttachedTo = component;
-    component.onAnchor(this._anchorCallback);
-
+    this._connect();
     return this;
   }
 
+  private _connect() {
```
Comment (bluong) [inline src/interactions/interaction.ts]: Unsure if I like the name.  My first response is ok `interaction._connect()` will connect to...?

That said, this is a private function and it alludes to what it's doing sorta somewhat ok.  Not a huge deal.

**Comment by jtlan** [comment]
Secondary issue filed at #2386.

**Pull request merged**